File: client/js/services/todo.service.js
(function() {
  'use strict'

  // サービス登録
  angular
    .module('app')
    .factory('TodoService', Service);

  // DI
  Service.$inject = ['$http'];

  // サービス
  function Service($http) {
    var baseUrl = 'http://localhost:8080/api/v1';

    return {
      'findAll': findAll,
      'find': find,
      'save': save,
      'remove': remove
    }

    // 全件取得
    function findAll(offset, limit) {
      var config = {
        params: {
          offset: offset,
          limit: limit
        }
      };
      return $http.get(baseUrl + '/todos', config);
    }

    // 1件取得
    function find(id) {
      var config = {
        params: {
          offset: offset,
          limit: limit
        }
      };
      return $http.get(baseUrl + '/todos/' + id, config);
    }

    // 保存
    function save(todo) {
      var data = todo;
      var config = {};
      return $http.post(baseUrl + '/todos', data, config);
    }

    // 削除
    function remove(id) {
      var config = {};
      return $http.delete(baseUrl + '/todos/' + id, config);
    }
  }
})();

File: server/src/main/java/com/example/service/TodoService.java
package com.example.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import com.example.entity.Todo;
import com.example.repository.TodoRepository;

/**
 * サービス
 */
@Service
@Transactional
public class TodoService {
	@Autowired
	private TodoRepository todoRepository;
	
	/**
	 * 全件取得
	 * @param offset
	 * @param limit
	 * @return
	 */
	@Transactional(readOnly = true)
	public Page<Todo> findAll(Integer offset, Integer limit) {
		return todoRepository.findAll(new PageRequest(offset, limit));
	}
	
	/**
	 * 1件取得
	 * @param id
	 * @return
	 */
	@Transactional(readOnly = true)
	public Todo findById(Integer id) {
		return todoRepository.findOne(id);
	}
	
	/**
	 * 保存
	 * @param todo
	 * @return
	 */
	public Todo save(Todo todo) {
		return todoRepository.save(todo);
	}
	
	/**
	 * 削除
	 * @param todo
	 * @return
	 */
	public void delete(Todo todo) {
		todoRepository.delete(todo);
	}
	
}
